### PR TITLE
Add symbolic links for libssl and libcrypto to Debian 9 builds.

### LINKED ISSUE #7598: New-PSSession on Linux to office365.com fails on Debian 9 due to missing symlinks for libssl and libcrypto
Client PSRP for Linux fails on a clean Debian 9 install of PowerShell due to the inability of libmi to resolve libssl and libcrypto.  While this worked as-is on Debian 8, Debian 9 fails unless symlinks for these two libraries are created in the $PSHOME directory.

To work around the problem, create the following symlinks in the PSHOME directory. Verify the version of ssl and crypto that are installed and adjust the references accordlingly.

```bash
sudo ln -s /usr/lib/x86_64-linux-gnu/libssl.so.1.0.2 libssl.so.1.0.0
sudo ln -s /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.2 libcrypto.so.1.0.0
```

Steps to reproduce
------------------

```powershell
$uri = 'https://outlook.office365.com/powershell-liveid'
$cred = Get-Credential
$session = New-PSSession -ConnectionUri $uri  -ConfigurationName Microsoft.Exchange -Credential $cred -Authentication Basic -AllowRedirection
```

Expected behavior
-----------------
The connection succeeds.

Actual behavior
---------------
The following error is reported:

This parameter set requires WSMan, and no supported WSMan client library was found. WSMan is either not installed or unavailable for this system.

### PR DESCRIPTION
**Pull request by dantraMSFT**
## PR Summary
On Debian 9, libmi cannot resolve libssl and libcrypto.  This change adds symbolic links to packaging and build to the $PSHOME directory to resolve the issue.

The fix was verified interactively connecting to [office 365](https://outlook.office365.com/powershell-liveid), importing the session, and ensuring Get-MailBox and Get-User succeed.

Fix https://github.com/PowerShell/PowerShell/issues/7598

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [X] Not Applicable or can only be tested interactively
    - **OR**
    - [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -610,6 +610,22 @@ Fix steps:
         }
     }
 
+    if ($Environment.IsDebian9)
+    {
+        # add two symbolic links to system shared libraries that libmi.so is dependent on to handle
+        # platform specific changes. This appears to be a change in Debian 9; Debian 8 did not need these
+        # symlinks.
+        if (!(Test-Path -Path "$publishPath/libssl.so.1.0.0"))
```
Comment (adityapatwardhan) [inline build.psm1]: The code for this is very similar to the code for RedHatFamily. Can we combine logic like this:

```PowerShell

if($Environment.IsRedHatFamily -or $Environment.IsDebian9){
if($Environment.IsRedHatFamily) {
    $sslTarget = "/lib64/libssl.so.10"
    $cryptoTarget = "/lib64/libcrypto.so.10"
} elseif ($Enviroment.IsDebian9) {
 …..
}
   if(-not (Test-Path …) {
   }

  if(-not (Test-Path …) {
  }
}

File: build.psm1 [hunk 2]
```diff
@@ -610,6 +610,22 @@ Fix steps:
         }
     }
 
+    if ($Environment.IsDebian9)
+    {
```
Comment (adityapatwardhan) [inline build.psm1]: Please follow the pattern of `{` being on the same line as `if`

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -1102,6 +1102,15 @@ function New-AfterScripts
         $AfterRemoveScript = [io.path]::GetTempFileName()
         $packagingStrings.UbuntuAfterInstallScript -f "$Link" | Out-File -FilePath $AfterInstallScript -Encoding ascii
         $packagingStrings.UbuntuAfterRemoveScript -f "$Link" | Out-File -FilePath $AfterRemoveScript -Encoding ascii
+
+        if ($Environment.IsDebian9)
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Please follow pattern for having `{` to be on the same line as `if`

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: build.psm1 (hunk 1 above)
Comment (dantraMSFT) [reply build.psm1]: fixed

File: build.psm1 (hunk 2 above)
Comment (dantraMSFT) [reply build.psm1]: fixed

File: tools/packaging/packaging.psm1 (hunk 3 above)
Comment (dantraMSFT) [reply tools/packaging/packaging.psm1]: fixed

File: build.psm1 [hunk 4]
```diff
@@ -597,16 +597,27 @@ Fix steps:
...
+    if ($Environment.IsRedHatFamily -or $Environment.IsDebian9) {
         # add two symbolic links to system shared libraries that libmi.so is dependent on to handle
         # platform specific changes. This is the only set of platforms needed for this currently
         # as Ubuntu has these specific library files in the platform and macOS builds for itself
         # against the correct versions.
+
+        if ($Environment.IsDebian9){
+            # NOTE: Debian 8 doesn't need these symlinks
+            $sslTarget = "/usr/lib/x86_64-linux-gnu/libssl.so.1.0.2"
+            $cryptoTarget = "/usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.2"
```
Comment (iSazonov) [inline build.psm1]: I wonder that it doesn't work in CoreFX https://github.com/janvorli/corefx/blob/13e2cc977713f0a5edc7a702583eeabb5643006d/src/Native/Unix/System.Security.Cryptography.Native/opensslshim.cpp#L43 (from https://github.com/dotnet/corefx/pull/27208)

Comment (dantraMSFT) [reply build.psm1]: I suspect it's because they are loading explicitly by version; libmi doesn't have version detection logic.

Comment (iSazonov) [reply build.psm1]: Do you mean that a root problem in libmi? if so it seems better to make a correction there than to receive surprises with each new linux distributive.

**Comment by TravisEz13** [comment]
@adityapatwardhan can you update your review?

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**